pkg/adaptor/xds/v3/route.go

```diff
@@ -262,3 +287,50 @@ func patchRoutesWithOriginalDestination(routes []*apisix.Route, origDst string)
 		}
 	}
 }
+
+// translateRouteAction translates envoy RouteAction to traffic-split plugins configs
+func (adaptor *adaptor) translateRouteAction(r *routev3.Route) (*apisix.Plugins, error) {
+	switch r.GetAction().(type) {
+	case *routev3.Route_Route:
+	default:
+		adaptor.logger.Infow("ignore unsupported route action",
+			zap.Any("action", r.GetAction()),
+		)
+		return nil, nil
+	}
+	action := r.GetAction().(*routev3.Route_Route).Route
+	switch action.GetClusterSpecifier().(type) {
+	case *routev3.RouteAction_WeightedClusters:
+	default:
+		adaptor.logger.Debugw("ignore single cluster",
+			zap.Any("cluster_specifier", action.GetClusterSpecifier()),
+		)
+		return nil, nil
+	}
+	clusters := action.GetClusterSpecifier().(*routev3.RouteAction_WeightedClusters).WeightedClusters.Clusters
```

Comment:
We don't need to use the traffic-split plugin if the number of clusters is `1`.

Reply:
Numbers of WeightedClusters from istio can't be 1, refer: http://github.com/istio/istio/blob/2dd7b6207f02cec8b42f4263ac197434f4ec9b4a/pilot/pkg/networking/core/v1alpha3/route/route.go#L496-L508

Reply:
Note here we don't consider the specific control plane role. As xDS itself doesn't limit it, we should check it.